### ISSUE TITLE
Suggestion: avoid using variables in construct IDs

### ISSUE THREAD

**Issue by rix0rrr** [ren-yamanashi/eslint-cdk-plugin#63]
### Feature Description

This library is a fantastic idea! I was going to suggest we create one ourselves, but I'm happy to see we don't need to do that anymore 😉 .

Here's a suggestion for a rule I was considering myself:

Avoid using variables in a construct ID

### Motivation

* In the best case, there is a lot of unnecessary repetition in the logical ID.
* In the worst case, the parameter changes and the the logical ID will be changed, leading to resource replacement.

This mistake is commonly made by users who don't understand the scoping of the construct ID, and are concerned that it might not be "unique enough", leading them to mix in all kinds of strings into the construct ID.

This rule is not trivial to implement, because sometimes there are valid reasons to mix a variable into a construct ID (for example, inside a `for` loop).

Maybe with a bit of data flow analysis the rule could be: don't mix parameter values into construct IDs.

### Invalid Code Examples

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, props: MyConstructProps) {
    super(scope, id);

    // ❌ Shouldn't use a parameter as a construct ID
    new Bucket(this, id);

    // ❌ Shouldn't combine a parameter into a template string
    new Bucket(this, `${id}Bucket`);

    // ❌ Shouldn't combine a parameter into any expression
    new Bucket(this, id + 'Bucket');

    // ❌ Shouldn't use a prop straight-up either
    new Bucket(this, `${prop.stage}Bucket`);
  }
}
```

### Valid Code Examples

```ts
class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, props: MyConstructProps) {
    super(scope, id);

    // ✅ Can use a literal
    new Bucket(this, 'Bucket');

    // ✅ Can use a loop variable
    for (const [key, value] of Object.entries(props.environments)) {
      new Bucket(this, `${key}Bucket`);
    }
  }
}
```

### Rule Type

problem (possible errors)

### Implementation Details

- [X] This rule is related to AWS CDK best practices
- [ ] I'm willing to submit a PR for this feature

**Comment by ren-yamanashi** [comment]
@rix0rrr 
Thank you for your wondaful idea!
I will incorporate this as a new rule, just like #62 that you suggested earlier!
